pkg/sql/schemachanger/scplan/internal/opgen/opgen_check_constraint.go

```diff
@@ -19,16 +19,48 @@ func init() {
 	opRegistry.register((*scpb.CheckConstraint)(nil),
 		toPublic(
 			scpb.Status_ABSENT,
+			to(scpb.Status_WRITE_ONLY,
+				emit(func(this *scpb.CheckConstraint) *scop.MakeAbsentCheckConstraintWriteOnly {
+					return &scop.MakeAbsentCheckConstraintWriteOnly{
+						TableID:               this.TableID,
+						ConstraintID:          this.ConstraintID,
+						ColumnIDs:             this.ColumnIDs,
+						Expression:            this.Expression,
+						FromHashShardedColumn: this.FromHashShardedColumn,
+					}
+				}),
+				// TODO (xiang): Add an operation that updates back-references in potentially
+				// affected types and sequences when implementing `add constraint`.
+			),
+			to(scpb.Status_VALIDATED,
+				emit(func(this *scpb.CheckConstraint) *scop.ValidateCheckConstraint {
+					return &scop.ValidateCheckConstraint{
+						TableID:      this.TableID,
+						ConstraintID: this.ConstraintID,
+					}
+				}),
+			),
 			to(scpb.Status_PUBLIC,
-				emit(func(this *scpb.CheckConstraint) *scop.NotImplemented {
-					return notImplemented(this)
+				emit(func(this *scpb.CheckConstraint) *scop.MakeValidatedCheckConstraintPublic {
+					return &scop.MakeValidatedCheckConstraintPublic{
+						TableID:      this.TableID,
+						ConstraintID: this.ConstraintID,
+					}
 				}),
 			),
 		),
 		toAbsent(
 			scpb.Status_PUBLIC,
+			to(scpb.Status_VALIDATED,
+				emit(func(this *scpb.CheckConstraint) *scop.MakePublicCheckConstraintValidated {
+					return &scop.MakePublicCheckConstraintValidated{
+						TableID:      this.TableID,
+						ConstraintID: this.ConstraintID,
+					}
+				}),
+			),
```

Comment:
Since this is revertible, please add a test case which covers rolling back a drop after PUBLIC -> VALIDATED (the rollback will exercise VALIDATED -> PUBLIC on the adding path), if nothing else.